apollo-router/src/query_planner/plan.rs

```diff
@@ -192,6 +193,64 @@ impl PlanNode {
         }
     }
 
+    /// Populate a Vec of QueryHashes representing Fetches in this plan.
+    ///
+    /// Do not include any operations which contain "requires" elements.
+    ///
+    /// Think about the impact of Defer/Subscription/Condition on this function. In
+    /// particular, Condition processing might be causing us to mis-report the number of fetches
+    /// and that would be a big problem. Perhaps we should Abort Batches which contain
+    /// Defer/Subscription/Condition....?
+    ///
+    pub(crate) fn query_hashes(&self, query_hashes: &mut Vec<Arc<QueryHash>>) {
```

Comment:
Note: I spoke with Derek about this and we don't need to accommodate Condition processing for this implementation of batching. I need to decide to convert this to return an error if Defer/Subscription/Condition is encountered (and make it iterative instead of recursive).

Reply:
done: be7da7877d57977d0f0425355b3a7380e48fdfa3